asynciterator.ts

```diff
@@ -1817,8 +1821,11 @@ export class UnionIterator<T> extends BufferedIterator<T> {
         // Pick the next source
         this._currentSource = (this._currentSource + 1) % this._sources.length;
         const source = this._sources[this._currentSource];
+        if (!source.source.readable && !source.requiresRead)
```

Comment:
Is there a reason reason why only having `!source.source.readable` is insufficient?

Reply:
It's to jumpstart the sub-iterators, I added this because I had issues with it in the past. If I remove it, all test still pass, so I guess it's not necessary? I guess an iterator should be created with readable set to true...

Reply:
> I guess an iterator should be created with readable set to true...

Yes, for iterators without autoStart, readable should be true.